text/0497-glue-l2-construct.md

```diff
@@ -0,0 +1,410 @@
+convenience methods working backwards from common use cases and default to
+recommended best practices.
+
+This RFC proposes updates to the L2 construct for Glue which will provide
+convenience features and abstractions for the existing
+[L1 (CloudFormation) Constructs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/AWS_Glue.html) building on the
+functionality already supported in the [@aws-cdk/aws-glue-alpha module](https://github.com/aws/aws-cdk/blob/v2.51.1/packages/%40aws-cdk/aws-glue/README.md).
+
+## Create a Glue Job
+
+The glue-alpha-module already supports three of the four common types of Glue
+Jobs: Spark (ETL and Streaming), Python Shell, Ray. This RFC will add the
+more recent Flex Job. The construct also implements AWS practice
+recommendations when creating a Glue Job such use of Secrets Management for
+Connection JDBC strings, Glue Job Autoscaling, least privileges in terms of
+IAM permissions and also sane defaults for Glue job specification (more details
+are mentioned in the below table).
+
+This RFC will introduce breaking changes to the existing glue-alpha-module to
+streamline the developer experience and introduce new constants and validations.
+The L2 construct will determine the job type by the job type and language
+provided by the developer, rather than having separate methods in every
+permutation that Glue jobs allow. As an opinionated construct, it will enforce
+best practices and not allow developers to create resources that use deprecated
+libraries and tool sets (e.g. deprecated versions of Python).
+
+### Spark Jobs
+
+1. **ETL Jobs**
+
+ETL jobs supports python and Scala language. ETL job type supports G1, G2, G4
+and G8 worker type default as G2, which customer can override. It wil default to
+the best practice version of ETL 4.0, but allow developers to override to 3.0.
+We will also default to best practice enablement the following ETL features:
+`—enable-metrics, —enable-spark-ui, —enable-continuous-cloudwatch-log.`
+You can find more details about version, worker type and other features in Glue's
+public documentation.
+
+```
+glue.ScalaSparkEtlJob(this, 'ScalaSparkEtlJob', {
+    script: glue.Code.fromBucket('bucket-name', 's3prefix/path-to-scala-jar'),
+    className: 'com.example.HelloWorld',
+});
+
+glue.pySparkEtlJob(this, 'pySparkEtlJob', {
+    script: glue.Code.fromBucket('bucket-name', 's3prefix/path-to-python-script'),
+});
+```
+
+Optionally, developers can override the glueVersion and add extra jars and a
+description:
+
+```
+glue.ScalaSparkEtlJob(this, 'ScalaSparkEtlJob', {
+   glueVersion: glue.GlueVersion.V3_0,
+   script: glue.Code.fromBucket('bucket-name', 's3prefix/path-to-scala-jar'),
+   className: 'com.example.HelloWorld',
+   extraJarsS3Url: [glue.Code.fromBucket('bucket-name', 'path-to-extra-scala-jar'),],
+   description: 'an example Scala Spark ETL job',
+   numberOfWorkers: 20
+});
+
+glue.pySparkEtlJob(this, 'pySparkEtlJob', {
+   jobType: glue.JobType.ETL,
```

Comment:
I don't think this prop makes sense here. It's implied in the name. Is it serving a purpose I might be missing?

Reply:
pySpark is known name for starting spark job written in python.

Reply:
But the fact that it's an ETL job is in the name so users shouldn't need to input the job type.